### PR TITLE
Migrate resource schema to github actions

### PR DESCRIPTION
**Pull request by jotompki**
*Issue #, if available:*

*Description of changes:* Migrates the resource schema repository to GitHub actions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      run: pip3 install --user pre-commit
```
Comment (jotompki) [inline .github/workflows/maven.yml:28]:
```suggestion
      uses: pre-commit/action@v2.0.0
```

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (PatMyron) [inline .github/workflows/maven.yml:15]: thoughts on adding `macos-latest` and `windows-latest`?

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      uses: pre-commit/action@v2.0.0
+    - name: Run pre-commit checks
```
Comment (jotompki) [inline .github/workflows/maven.yml:29]:
```suggestion
    with:
        extra_args: --all-files 
```

**Comment by PatMyron** [comment]
Thoughts on testing with more `java`/`python` versions?

**Review by PatMyron — APPROVED** [review]

File: .github/workflows/maven.yml (hunk 2 above)
Comment (jotompki) [reply .github/workflows/maven.yml:15]: this is a java lib jvm should be os agnostic

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      uses: pre-commit/action@v2.0.0
+    with:
```
Comment (jotompki) [inline .github/workflows/maven.yml:29]:
```suggestion
        with:
```

File: .github/workflows/maven.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      uses: pre-commit/action@v2.0.0
+        with:
```
Comment (jotompki) [inline .github/workflows/maven.yml:29]:
```suggestion
      with:
```

File: .github/workflows/maven.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      uses: pre-commit/action@v2.0.0
+      with:
+        extra_args: --all-files 
+      run: pre-commit run --all-files --verbose
```
Comment (jotompki) [inline .github/workflows/maven.yml:31]:
```suggestion
```

**Comment by jotompki** [comment]
> Thoughts on testing with more `java`/`python` versions?

Python is just really used to run pre-commit; there's no python code in the repo. 

We can look into testing with more java versions, but we are only targeting 1.8 in the pom

File: .github/workflows/maven.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (PatMyron) [inline .github/workflows/maven.yml:10]: also wouldn't mind running all these Github actions on all branches:
```suggestion
on: [push, pull_request
```

**Review by sungkkim — APPROVED** [review]

**Comment by PatMyron** [comment]
We should automate releases using Github Actions as well

**Comment by PatMyron** [comment]
Could get rid of the Travis badge:
https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/blob/b4d8d2bc2badcd7019c4e445090a0617a110d9e1/README.md#L3

File: .github/workflows/maven.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        java-version: 1.8
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: install pre-commit
+      uses: pre-commit/action@v2.0.0
+      with:
+        extra_args: --all-files
```
Comment (PatMyron) [inline .github/workflows/maven.yml:30]: What's the benefit of [`pre-commit/action/`](https://github.com/pre-commit/action/) vs. installing/running `pre-commit` normally?

**Pull request merged**